tests/end-to-end/general/idempotent-apply.bats

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env bats
+
+# bats file_tags=idempotent-apply
+
+setup_file() {
+  # for --separate-stderr argument on 'run'
+  bats_require_minimum_version 1.11.1
+
+  export BATS_NO_PARALLELIZE_WITHIN_FILE=true
+  export CK8S_AUTO_APPROVE="true"
+
+  load "../../bats.lib.bash"
+  load_assert
+
+  load_common "git.bash"
+
+  if git.is_modified "$CK8S_CONFIG_PATH"; then
+    fail "Fatal: CK8S_CONFIG_PATH (${CK8S_CONFIG_PATH}) is tracked in a git repository and has uncommitted changes. Please commit your changes and be mindful that these tests will 'apply' all of the application stacks in both the SC and the WC clusters."
+  fi
```

Comment:
The only concern I have is that a config path may include a lot more than just the Apps config, so this might become a bit too overprotective but it's a simpler check.